task/buildah-multiplatform/0.1/buildah-multiplatform.yaml

```diff
@@ -0,0 +1,108 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: "docker"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "image-build, appstudio, hacbs"
+  name: buildah-multiplatform
+spec:
+  description: |-
+    Buildah Multiplatform takes existing images and stiches them together into a multi platform image.
+  params:
+  - description: Reference of the image buildah will produce.
+    name: IMAGE
+    type: string
+  - default: registry.access.redhat.com/ubi9/buildah:9.0.0-19@sha256:c8b1d312815452964885680fc5bc8d99b3bfe9b6961228c71a09c72ca8e915eb
+    description: The location of the buildah builder image.
+    name: BUILDER_IMAGE
+    type: string
+  - default: "true"
+    description: Verify the TLS on the registry endpoint (for push/pull to a non-TLS registry)
+    name: TLSVERIFY
+    type: string
+  - name: COMMIT_SHA
+    description: The image is built from this commit.
+    type: string
+    default: ""
+  - name: IMAGES
+    description: List of images that are to be merged into the multi platform image
+    type: array
+  - default: ""
+    description: Delete image tag after specified time. Empty means to keep the image tag. Time values could be something like 1h, 2d, 3w for hours, days, and weeks, respectively.
+    name: IMAGE_EXPIRES_AFTER
+    type: string
+  results:
+  - description: Digest of the image just built
+    name: IMAGE_DIGEST
+  - description: Image repository where the built image was pushed
+    name: IMAGE_URL
+  stepTemplate:
+    env:
+    - name: BUILDAH_FORMAT
+      value: oci
+    - name: STORAGE_DRIVER
+      value: vfs
+    - name: IMAGE
+      value: $(params.IMAGE)
+    - name: TLSVERIFY
+      value: $(params.TLSVERIFY)
+    - name: IMAGE_EXPIRES_AFTER
+      value: $(params.IMAGE_EXPIRES_AFTER)
+  steps:
+  - image: quay.io/redhat-appstudio/buildah:v1.28
+    name: build
+    resources:
+      limits:
+        memory: 4Gi
+        cpu: 2
+      requests:
+        memory: 512Mi
+        cpu: 250m
+    env:
+    - name: COMMIT_SHA
+      value: $(params.COMMIT_SHA)
+    args: ["$(params.IMAGES[*])"]
+    script: |
+      
+      
+      # Fixing group permission on /var/lib/containers
+      chown root:root /var/lib/containers
+
+      sed -i 's/^\s*short-name-mode\s*=\s*.*/short-name-mode = "disabled"/' /etc/containers/registries.conf
+
+      # Setting new namespace to run buildah - 2^32-2
+      echo 'root:1:4294967294' | tee -a /etc/subuid >> /etc/subgid
+
+      buildah manifest create $IMAGE
+      for i in $@
+      do
+        unshare -Uf --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -- buildah pull $i
+        unshare -Uf --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -- buildah manifest add $IMAGE $i
+      done
```

Comment:
Can we have a follow-up story to improve this? Pulling the entire image just to add its manifest to an OCI index is unnecessary

Maybe `crane` does it better? https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane_index_append.md

And so does OSBS, but in Python

Reply:
I have renamed this task to `build-image-manifest`. We will likely replace buildah at some point the name should  not be tied to the implementation.